### PR TITLE
Add yarn support, remove typings, and fix AoT issue

### PR DESCRIPTION
**Pull request by jeffbcross**

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -20,7 +20,6 @@
   },
   "files": [
     "src/test-root.ts",
-    "typings/index.d.ts",
```
Comment (katowulf) [inline tsconfig.json]: ♡

**Comment by katowulf** [comment]
Out of curiosity, what is yarn.lock for? It's 5k lines??

**Comment by katowulf** [comment]
LGTM!

**Comment by jeffbcross** [comment]
@katowulf yarn.lock is the version-locking file that [yarn](https://github.com/yarnpkg/yarn) generates. It's in lieu of npm-shrinkwrap, so we can lock down to exact versions locally and on Travis when we use `yarn install`. This makes reproducing build or test failures easier between environments.

**Pull request merged**